bread/layout/components/modal.py

```diff
@@ -81,6 +81,13 @@ def __init__(
             **attributes,
         )
 
+    def attach_to_trigger(self, trigger: hg.BaseElement):
+        """
+        Attaches this modal to a trigger element and returns a wrapper element
+        """
+        trigger.attributes.update(self.openerattributes)
```

Comment:
For me, attribute updates (especially dicts) are never a good thing. I find immutable objects much easier to reason about. 
Therefore I would prefer a shortcut function that takes everything that is needed to create the modal _and_ the trigger and creates both. In this specific case the overhead that is needed seems significant, but I think immutability wins in the long run.

Reply:
btw this shortcut function could also take care of adding the "asajax" parameter and maybe other stuff.

Reply:
We are using the "construct" of modal with a trigger in a number of places now. And still it is not simple to create even the most trivial working instance of this construct. I think we should make this very easy. So maybe we can just have a shortcut "modal_with_trigger(path_of_view: str, [optional params])" that creates a "edit"-button and a modal with the title "edit" and a lot of optional parameters to customize where needed.

Reply:
the title of the modal can also be more clever than just "edit", it could be "Edit <automatically figure out name of the model>"

Reply:
I am not sure what exactly you would like to change. You would create a new dict and assign it to ```trigger.attributes```? Then we still mutated the trigger element, so we still have state change. Yes, we could avoid to change the state of ```attributes``` but it wouldnt' make sense because dict is a mutable object.

Maybe some background here (because ```attributes``` is coming from htmlgenerator): I decided to write htmlgenerator with mutable datastructures. I did some experiments with immutable datastructures first. But it turned out that the code would in many cases be more complicated to write, especially the API and construction of tree elements. Not necessarily because of the mutability per se but because Python is in most cases designed around the assumption of mutable objects.

I agree that immutability can be an advantage in many cases. But I would argue that it should not be imperative just for the sake of it. The reason for my thinking is: Mathematics is immutable but the real world is mutable. Therefore we will not be able to "factor out immutability" in many cases.

Reply:
> I am not sure what exactly you would like to change. You would create a new dict and assign it to trigger.attributes?

No, that's not what I meant. I just prefer it if the trigger is created with the right attributes from the beginning. Assigning a new dict would be just as ugly and wouldn't help at all because no one holds a "pointer" to the dict anyway.

So I am actually just saying that I in principle prefer it the way we are already doing it at other places, but I agree that we should streamling the process of creating triggers and button and I am proposing another way of doing this.